components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/internal/WorkflowImplServiceComponent.java

```diff
@@ -133,14 +136,16 @@ private void addDefaultBPSProfile() {
         try {
             WorkflowImplService workflowImplService = WorkflowImplServiceDataHolder.getInstance().getWorkflowImplService();
             BPSProfile currentBpsProfile = workflowImplService.getBPSProfile(WFImplConstant.DEFAULT_BPS_PROFILE_NAME, MultitenantConstants.SUPER_TENANT_ID);
-            String url = IdentityUtil.getServerURL(WorkflowImplServiceDataHolder.getInstance().getConfigurationContextService().getServerConfigContext().getServicePath(), true, true);
+            String url = buildDefaultBPSServerUrl();
```

Comment:
What if we use the serviceURLBuilder here directly and catch the exception?

Reply:
we cannot use serviceURLBuilder because internally it is constructing tenant qualified URLs when tenant qualified feature enabled. Embedded BPS is not deployed per tenant.